src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,485 @@
+            }
+        }
+
+        private string PackageInstallPath { get; set; }
+
+        public void GoToWebsite()
+        {
+            dynamoViewModel.PackageManagerClientViewModel.GoToWebsite();
+        }
+
+        /// <summary>
+        /// Install Dynamo Package
+        /// </summary>
+        /// <param name="downloadPath"></param>
+        private void InstallPackage(string downloadPath)
+        {
+            var firstOrDefault = Model.LocalPackages.FirstOrDefault(pkg => pkg.ID == DownloadRequest.asset_id.ToString());
+            if (firstOrDefault != null)
+            {
+                var dynModel = dynamoViewModel.Model;
+                try
+                {
+                    firstOrDefault.UninstallCore(dynModel.CustomNodeManager, dynamoViewModel.Model.GetPackageManagerExtension().PackageLoader, dynModel.PreferenceSettings);
+                }
+                catch
+                {
+                    MessageBox.Show(String.Format(Resources.MessageFailToUninstallPackage,
+                        dynamoViewModel.BrandingResourceProvider.ProductName,
+                        DownloadRequest.asset_name.ToString()),
+                        Resources.UninstallFailureMessageBoxTitle,
+                        MessageBoxButton.OK, MessageBoxImage.Error);
+                }
+            }
+
+            var settings = dynamoViewModel.Model.PreferenceSettings;
+            PackageDownloadHandle packageDownloadHandle = new PackageDownloadHandle();
+            packageDownloadHandle.Name = DownloadRequest.asset_name;
+            packageDownloadHandle.Done(downloadPath);
+
+            string installedPkgPath = string.Empty;
+            if (packageDownloadHandle.Extract(dynamoViewModel.Model, this.PackageInstallPath, out installedPkgPath))
+            {
+                var p = Package.FromDirectory(installedPkgPath, dynamoViewModel.Model.Logger);
+                p.ID = DownloadRequest.asset_id;
+                Application.Current.Dispatcher.Invoke((Action)(() =>
+                {
+                    dynamoViewModel.Model.GetPackageManagerExtension().PackageLoader.Load(p);
+                }));
+                packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Installed;
+                this.PackageInstallPath = string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Download and Install Dynamo Package
+        /// </summary>
+        /// <param name="pkg"></param>
+        public void DownloadAndInstall(string pkg)
+        {
+            string[] temp = pkg.Split(',');
+            PackageManagerRequest req = new PackageManagerRequest("assets/" + temp[0], Method.GET);
+            ResponseWithContentBody<dynamic> response = DPClient.ExecuteAndDeserializeDynamoRequest(req);
+            DownloadRequest = response.content;
+
+            PackageManagerRequest fileReq = new PackageManagerRequest(@"files/download?file_ids=" + temp[1] + "&asset_id=" + temp[0], Method.GET, true);
```

Comment:
Can we use string builder or string.Format to build this string? Also move the constant string as const.